staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/helpers.go

```diff
@@ -51,6 +53,70 @@ func SortedByGroupAndVersion(servers []*APIService) [][]*APIService {
 	return ret
 }
 
+type versionType int
+
+const (
+	// Bigger the version type number, higher priority it is
+	versionTypeAlpha versionType = 0
+	versionTypeBeta  versionType = iota
+	versionTypeGA    versionType = iota
+)
+
+func parseKubeVersion(v string) (majorVersion int, vType versionType, minorVersion int, ok bool) {
+	var err error
+	re := regexp.MustCompile("^v([\\d]+)(?:(alpha|beta)([\\d]+))?$")
+	submatches := re.FindStringSubmatch(strings.ToLower(v))
```

Comment:
don't lowercase the string, should match exactly

Reply:
what should we do if the version is `v1Beta1`? options: 1. consider it out of format, 2. consider it in-format and put it before `v1beta1`?

Reply:
out of format